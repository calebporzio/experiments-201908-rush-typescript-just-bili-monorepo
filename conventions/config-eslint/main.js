/**
 * JavaScript/ECMAScript only ruleset.
 *
 * Since JavaScript does not understand TypeScript, we can't put
 * TypeScript rules alongside JavaScript, hence this separation.
 */
module.exports = {
  plugins: [
    // https://www.npmjs.com/package/eslint-plugin-promise
    "eslint-plugin-promise"
  ]
};
