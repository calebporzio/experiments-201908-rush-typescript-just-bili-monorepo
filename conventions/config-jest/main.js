const path = require("path");

/**
 * https://github.com/microsoft/just/blob/master/scripts/jest.config.js
 *
 * @type {import('jest').DefaultOptions}
 */
module.exports = {
  clearMocks: true,
  expand: true,
  forceExit: true,
  globals: {
    "ts-jest": {
      tsConfig: path.resolve(process.cwd(), "tsconfig.json"),
      packageJson: path.resolve(process.cwd(), "package.json")
    }
  },
  preset: "ts-jest",
  roots: ["<rootDir>/src"],
  testEnvironment: "node",
  testPathIgnorePatterns: ["/lib/", "/node_modules/"],
  testRegex: "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
  verbose: true,
  verbose: true
};
