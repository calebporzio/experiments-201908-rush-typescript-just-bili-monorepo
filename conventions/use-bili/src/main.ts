import { Config } from 'bili';
import { plugins as initPlugins } from './plugins';
import { input as initInput, BiliInputDescriptor } from './input';

// import { default as bundlingHelpersMain } from '@frontend-bindings/tools-bundling-helpers';
// import { resolve, dirname } from 'path';

/**
 * @public
 * @param {NodeJS.ProcessEnv} env NodeJS Runtime process.env value
 */
export const main = (
  inputArg: string | BiliInputDescriptor = 'src/index.ts',
) => (env: NodeJS.ProcessEnv = {}): Config => {
  const plugins = initPlugins(env);
  const input = initInput(inputArg);

  /**
   * Unfinished work.
   * ... should encapsulate what ../bili.config.ts does
   */
  // const { cwd } = process || {};
  // if (cwd) {
  //   const baseDir = resolve(dirname(cwd()));
  //   const bundle = bundlingHelpersMain(baseDir);
  //   console.log('foo', { bundle, baseDir });
  // }

  const out: Config = {
    input,
    plugins,
  };

  return out;
};
