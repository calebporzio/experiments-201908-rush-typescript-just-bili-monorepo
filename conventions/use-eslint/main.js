const ecmascript = require('./ecmascript');
const typescript = require('./typescript');

const plugins = [...ecmascript.plugins, ...typescript.plugins];
const main = {
  ...ecmascript,
  ...typescript,
  plugins,
};

module.exports = main;
