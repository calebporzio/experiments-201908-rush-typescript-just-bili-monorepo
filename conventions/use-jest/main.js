const { resolve, join, normalize } = require('path');

const jestModuleDir = __dirname;
const prettierPath =
  '/node_modules/@frontend-bindings/conventions-use-prettier/bin/prettier';

const processEnvKeys = Object.keys(process.env);
const isCI = processEnvKeys.includes('CI_SERVER');
const verbose = !isCI;

/**
 * https://github.com/microsoft/just/blob/master/scripts/jest.config.js
 *
 * @type {import('jest').DefaultOptions}
 */
const main = {
  clearMocks: true,
  expand: true,
  forceExit: true,
  prettierPath: resolve(normalize(join(__dirname, prettierPath))),
  globals: {
    'ts-jest': {
      tsConfig: resolve(process.cwd(), 'tsconfig.json'),
      packageJson: resolve(process.cwd(), 'package.json'),
    },
  },
  preset: 'ts-jest',
  roots: ['<rootDir>/src'],
  setupFilesAfterEnv: ['jest-expect-message'],
  testEnvironment: 'node',
  testPathIgnorePatterns: ['/dist/', '/node_modules/'],
  testRegex: '__tests__/.*\\.(test|spec)\\.(jsx?|tsx?)$',
  transform: {
    // See comment in ecmascript-transformer.js
    // '\\.jsx?$': path.join(jestModuleDir, 'ecmascript-transformer.js'),
    '\\.(ts|js)x?$': 'ts-jest',
  },
  verbose,
};

if (isCI) {
  // If we do not superseed the reporters, we might get errors when
  main.reporters = [join(jestModuleDir, 'reporter.js')];
}

// console.log('use-jest', { isCI, reporters: [...main.reporters], verbose })

module.exports = main;
