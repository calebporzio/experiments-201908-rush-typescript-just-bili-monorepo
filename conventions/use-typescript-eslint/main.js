// See https://github.com/microsoft/tsdoc/blob/master/playground/.eslintrc.js#L2
// This is a workaround for https://github.com/eslint/eslint/issues/3458
require('@microsoft/eslint-config-scalable-ts/patch-eslint6');

const base = require('@frontend-bindings/conventions-use-eslint');

/**
 * TypeScript ruleset.
 *
 * https://eslint.org/blog/2019/01/future-typescript-eslint
 *
 * @type {import('@types/eslint').Linter.Config}
 */
module.exports = {
  ...base,
  // https://www.npmjs.com/package/@typescript-eslint/parser
  parser: '@typescript-eslint/parser',
  extends: ['plugin:@typescript-eslint/recommended'],
  plugins: [
    ...(base.plugins || []),
    // https://www.npmjs.com/package/@typescript-eslint/eslint-plugin
    '@typescript-eslint/eslint-plugin',
  ],
  overrides: [
    {
      files: ['*.test.js', '*.test.ts', '*.test.tsx', '*.test.jsx'],
      globals: {
        it: 'readonly',
        expect: 'readonly',
        test: 'readonly',
        describe: 'readonly',
      },
    },
  ],
};
