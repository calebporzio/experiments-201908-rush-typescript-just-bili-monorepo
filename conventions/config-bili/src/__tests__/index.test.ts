import main from '..';

describe('main', () => {
  it('Snapshot sample', () => {
    const subject = main('foo/bar/bazz/buzz/bizz.ts')(process.env);
    expect(subject).toMatchSnapshot();
  });
});
