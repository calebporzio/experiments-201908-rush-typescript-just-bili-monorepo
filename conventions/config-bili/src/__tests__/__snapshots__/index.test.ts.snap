// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main Snapshot sample 1`] = `
Object {
  "input": Object {
    "index": "foo/bar/bazz/buzz/bizz.ts",
  },
  "plugins": Object {
    "babel": Object {
      "presets": Array [
        Array [
          "@babel/preset-env",
          Object {
            "corejs": 3,
            "debug": false,
            "useBuiltIns": "usage",
          },
        ],
      ],
    },
  },
}
`;
